src/base/get_version.sh

```diff
@@ -89,6 +89,7 @@ fi
 # Overwrite ./version.h with the temporary file if they are different.
 if ! cmp -s $temp version.h; then
   cp $temp version.h
+  sed -i 's/\r//g' version.h
```

Comment:
try to avoid sed -i (or sed altogether)

Reply:
actually, try to figure out where the linefeed character comes from originally (does 'echo' add it?) and I'd like to know what problems it causes.

Reply:
but there would be '\r' in the 'version.h' file since it is on windows. should I leave it to users to delete '\r' themselves?

Reply:
kaldi itself does not have '\r'. this character is automatically generated when I download kaldi to a windows machine. '\r' in windows means 'end of line' like '\n' in Unix and the Unix environment(cygwin) cannot recognize it.

Reply:
Actually, I think the logic of your change is wrong, your code will cause
the `if ! cmd ` test always succeed, so you will always replace the file --
you should modify the $temp file instead